microsoft/testsuites/virtualclient/virtualclient.py

```diff
@@ -0,0 +1,92 @@
+from typing import List, Optional, cast
+
+from lisa import (
+    Logger,
+    RemoteNode,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.features import Disk
+from lisa.operating_system import Debian
+from lisa.testsuite import TestResult
+from lisa.tools import VcRunner, VcTargetInfo
+
+
+@TestSuiteMetadata(
+    area="virtual_client",
+    category="performance",
+    description="""
+        This test suite runs the performance test cases with Virtual Client.
+    """,
+)
+class VirtualClient(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            This test is to run redis workload testing with Virtual Client.
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            supported_os=[Debian],
+            min_count=2,
+        ),
+    )
+    def perf_vc_redis(self, log: Logger, result: TestResult) -> None:
+        self._run_work_load(result=result, profile_name="PERF-REDIS", roles=["client"])
+
+    @TestCaseMetadata(
+        description="""
+            This test is to run PostgreSQL workload testing with Virtual Client.
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            supported_os=[Debian],
+            min_count=2,
+        ),
+        timeout=3000,
+    )
+    def perf_vc_postgresql(self, log: Logger, result: TestResult) -> None:
+        environment = result.environment
```

Comment:
the environment is supported at the test method like node, you don't need to get it from test result.